theme/tests/test_functional.py

```diff
@@ -1,48 +1,116 @@
+    #def test_show_login_link_mobile(self):
+    #    """The login link and form should not be displayed on the home page on mobile"""
+    #    self.mobile_drvr.get(self.live_server_url)
+    #    mobile_login_link = self.mobile_drvr.find_element_by_id('signin-menu')
+    #    self.assertFalse(mobile_login_link.is_displayed(),
+    #                     'Login link should not be visible on mobile.')
 
     def test_show_login_link_desktop(self):
         """The login link but not form should be displayed on the home page"""
-        self.desktop_browser.get(self.live_server_url)
-        desktop_login_link = self.desktop_browser.find_element_by_id('signin-menu')
+        self.drvr.get(self.live_server_url)
+        desktop_login_link = self.drvr.find_element_by_id('signin-menu')
         self.assertTrue(desktop_login_link.is_displayed(),
                         'Login link should be visible on desktop.')
+
+    def test_register_account(self):
+        """The login link but not form should be displayed on the home page"""
+        with self.settings(CSP_DICT = {'default-src': 'unsafe-eval'}):
+            self.drvr.get(self.live_server_url)
+            self.drvr.find_element_by_class_name('glyphicon-log-in').click()
+            self.drvr.find_element_by_id('id_username').click()
+            self.assertTrue('accounts/login' in self.drvr.current_url)
+            self.drvr.find_element_by_link_text('join HydroShare').click()
+            form = self.drvr.find_element_by_id('form-signup')
+            self.assertTrue(form.is_displayed())
+
+
+    def _login_helper(self, login_name, user_password):
+        with self.settings(CSRF_COOKIE_SECURE = False,
+                           SESSION_COOKIE_SECURE = False,
+                           CSP_DICT = {'default-src': 'unsafe-eval'}):
+            # home page: click to login
+            self.drvr.get(self.live_server_url)
+            self.drvr.find_element_by_class_name('glyphicon-log-in').click()
+
+            # login page: fill login form
+            username_field = self.drvr.find_element_by_id('id_username')
+            password_field = self.drvr.find_element_by_id('id_password')
+            submit = self.drvr.find_element_by_xpath("//input[@type='submit']")
+
+            username_field.send_keys(login_name, keys.Keys.TAB)
+            self.assertEquals(self.drvr.switch_to.active_element, password_field)
+            password_field.send_keys(user_password, keys.Keys.TAB)
+            self.assertEquals(self.drvr.switch_to.active_element, submit)
+            submit.send_keys(keys.Keys.ENTER)
+            self.drvr.save_screenshot('enter_pressed' + login_name + '.png')
+
+            # home page: returned after successful login
+            self.assertTrue('Successfully logged in' in self.drvr.page_source)
+            self.assertEqual(self.live_server_url + "/", self.drvr.current_url)
+            profile = self.drvr.find_element_by_id('profile-menu')
```

Comment:
We're hanging a lot of these tests on ids - this could be brittle and cause issues later of a FE dev and goes in and change stuff. We might want to consider using the same selectors that Django forms, the CSS, or the client-side JS uses, in this case the name attribute on the form.

Conversely, if we DO add ids for testing, we need a way of marking them as "sacred" so the FE dev is warned not to mess with it. Something like "THISISFORTESTINGDONOTTOUCH-profile-menu" but maybe a little less heavy handed :)

All that said, this PR is :+1: :100: :1st_place_medal: and I'm thoroughly psyched that we have tests like this in place at all.